### PR TITLE
Rework fsl-eula-unpack.bbclass for reuse

### PR DESCRIPTION
**Pull request by thochstein**
We have an internal use for the license information calculated by class.

### CODE REVIEW

File: classes/fsl-eula-unpack.bbclass [hunk 1]
```diff
@@ -128,6 +128,8 @@ python fsl_bin_do_unpack() {
         bb.note("Handling file '%s' as a Freescale EULA-licensed archive." % url.basename)
         cmd = "sh %s --auto-accept --force" % (url.localpath)
         bb.fetch2.runfetchcmd(cmd, d, quiet=True, workdir=rootdir)
+    if not found:
+        bb.warn("The package includes 'fsl-eula-unpack.bbclass' but does not mark any URL in SRC_URI with the option 'fsl-eula=true'")
```
Comment (otavio) [inline classes/fsl-eula-unpack.bbclass:154]: `bb.fatal` might enforce it.

File: classes/fsl-eula-unpack.bbclass [hunk 2]
```diff
@@ -157,11 +157,21 @@ python fsl_bin_do_unpack() {
...
+    (layer_license, licenses, md5sums, found_layer_licenses, found_package_licenses) = findNxpEulaLicenses(d)
+    if not found_layer_licenses:
+        bb.fatal("The Freescale layer EULA '%s' is not listed in LIC_FILES_CHKSUM '%s'."
+                 % (layer_license, licenses))
+    if not found_package_licenses:
+        bb.fatal("A valid package EULA with md5sum in %s was not found in LIC_FILES_CHKSUM '%s'."
+                 % (md5sums.split(), licenses))
+}
+
+def findNxpEulaLicenses(d):
```
Comment (otavio) [inline classes/fsl-eula-unpack.bbclass:169]: Python uses `find_nxp_eula_licenses(d):` so the snake_case.

**Review by otavio — CHANGES_REQUESTED** [review]

File: classes/fsl-eula-unpack.bbclass (hunk 2 above)
Comment (thochstein) [reply classes/fsl-eula-unpack.bbclass:169]: Haha, old habit.

File: classes/fsl-eula-unpack.bbclass [hunk 3]
```diff
@@ -171,14 +181,8 @@ python fsl_bin_do_unpack() {
...
-            found_package_license = True
-    if not found_layer_license:
-        bb.fatal("The Freescale layer EULA '%s' is not listed in LIC_FILES_CHKSUM '%s'."
-                 % (layer_license, licenses))
-    if not found_package_license:
-        bb.fatal("A valid package EULA with md5sum in %s was not found in LIC_FILES_CHKSUM '%s'."
-                 % (md5sums.split(), licenses))
-}
+            found_package_licenses += license
+    return (layer_license, licenses, md5sums, found_layer_licenses, found_package_licenses)
```
Comment (otavio) [inline classes/fsl-eula-unpack.bbclass:188]: only the first two arguments are used. It could reduce the number of tuple items.

**Review by otavio — CHANGES_REQUESTED** [review]

File: classes/fsl-eula-unpack.bbclass (hunk 3 above)
Comment (thochstein) [reply classes/fsl-eula-unpack.bbclass:188]: Not sure what's best. In my reuse case, I want the other values as well, so it does reduce a few lines of duplication.

Comment (otavio) [reply classes/fsl-eula-unpack.bbclass:188]: That's fine then.

**Pull request merged**